src/python/content.cpp

```diff
@@ -817,6 +826,10 @@ builder_fromiter(ak::ArrayBuilder& self, const py::handle& obj) {
   else if (py::isinstance<py::float_>(obj)) {
     self.real(obj.cast<double>());
   }
+  // FIXME: should be something like py::isinstance<py::complex_>(obj)
+  else if (obj.attr("__class__").attr("__name__").cast<std::string>() == "complex") {
```

Comment:
```cpp
bool is_complex(py::object obj) {
    static py::object cmplx = py::module_::import("builtins").attr("complex");
    return py::isinstance(obj, cmplx);
}
```

Gives me:

```python
>>> import cppimport, setuptools, numpy
>>> cplx_check.is_complex(1)
False
>>> cplx_check.is_complex(1.2j)
True
>>> cplx_check.is_complex(numpy.complex128(1j))
True
```

You'll need to fix the above import to also support Python 2.

Reply:
In Python 2, it's

```c++
py::module_::import("__builtin__").attr("complex")
```

(singular, with double-underscores). It should be possible to try to import both modules—only one of them will exist.

Reply:
You can try `"builtins"` first, then fall back to `"__builtin__"` by catching `py::import_error` (https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html).

Reply:
I kept wavering on the necessity of this one: it works without it, but if a class happens to be named "`complex`" while not being the built-in Python `complex` type, then this would pass when it shouldn't. I just did the fix and will be pushing it soon.

Reply:
An if would be quite a lot cheaper than an exception, and a reader can instantly tell which one is Python 2 and which is Python 3. In fact, you can make this a compile-time `#if` and simply not do the check at runtime at all.

Reply:
`#if PY_VERSION_HEX < 0x03000000` or `#if PY_MAJOR_VERSION < 3`

Reply:
That would work, too. I was about to say, "I'm not concerned about the performance in the Python <--> C++ interface," but this one happens to be in the `from_iter` loop, which can potentially be big. Okay, since I haven't merged the PR yet, I'll do it.

Reply:
Also, a Python 2.7 test failed; apparently the ImportError wasn't getting caught. I just checked in a fix and I'm manually testing Python 2.7 as well.